src/aap_eda/api/views/decision_environment.py

```diff
@@ -117,3 +110,33 @@ def retrieve(self, request, pk):
                 decision_environment.data
             ).data
         )
+
+    @extend_schema(
+        description="Delete a decision environment by id",
+        responses=status.HTTP_204_NO_CONTENT,
+        parameters=[
+            OpenApiParameter(
+                name="force",
+                description="Force deletion if there are dependent objects",
+                required=False,
+                type=bool,
+            )
+        ],
+    )
+    def destroy(self, request, *args, **kwargs):
+        instance = self.get_object()
+        force_delete = request.query_params.get("force", "false").lower()
+
+        activations_exist = models.Activation.objects.filter(
+            decision_environment_id=instance.id
+        ).exists()
+
+        if activations_exist and force_delete not in ["true", "1", "yes"]:
+            raise api_exc.Conflict(
+                "Decision Environment is being used by Activations "
+                "and cannot be deleted. If you want to force delete, "
+                "please add /?force=True query param."
```

Comment:
@Dostonbek1 Should this be lowercase or uppercase True?

Reply:
It doesn't really matter. On line 128, we deal with different variations of the param string. So it can be `True`, `true`, `TRUE`, etc. and we do `.lower()` for comparison.